lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,54 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after 20  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+final Matcher doesNotComplete = const _DoesNotComplete(20);
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after [timesToPump]  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+Matcher doesNotCompleteAfter(int timesToPump) =>
```

Comment:
I'm not sure it's worth making this a separate function. I think it's pretty unlikely that users will want to have fine-grained control over how long the test runs, and if they do they can just write `expect(pumpEventQueue(n), completes)`.